.changeset/spicy-hats-travel.md

```diff
@@ -0,0 +1,29 @@
+---
+"effect": minor
+---
+
+add RcMap module
+
+An `RcMap` can contain multiple reference counted resources that can be indexed
+by a key. The resources are lazily acquired on the first call to `get` and
+released when the last reference is released.
+
+```ts
+import { Effect, RcMap } from "effect";
+
+Effect.gen(function* () {
+  const map = yield* RcMap.make({
+    lookup: (key: string) =>
```

Comment:
It may be worth noting both here and in the relevant JSDocs that they `key` of `lookup` will be compared by `Hash`.